python-checks/src/test/resources/checks/confusingWalrus.py

```diff
@@ -0,0 +1,36 @@
+# using an assignment expression (:=) as an assignment statement (=) is more explicit
+(v := f(p))  # Noncompliant
+v0 = (v1 := f(p))  # Noncompliant
+
+# using an assignment expression in a function call when keyword arguments are also used.
+func(a=(b := f(p)))  # Noncompliant
+func(a := f(p), b=2)  # Noncompliant
+func(a := f(p), 3) # OK
+func(a := f(p), (3, 4), *args) # OK
```

Comment:
What about the following?
```
def f(a=(x:=2)+42, b=1): ...
```

Reply:
We will raise issue every time an assignment expression is used in a function definition where parameter default values are also present, as this can get difficult to read.